src/core/ledger/grainAllocation.js

```diff
@@ -288,3 +342,12 @@ export const allocationParser: P.Parser<Allocation> = P.object({
   id: uuidParser,
   receipts: P.array(grainReceiptParser),
 });
+
+export opaque type Discount = number;
```

Comment:
by making this opaque, it means that other modules won't be allowed to use it like a number (e.g. calling `.toFixed()`), which is a little too restrictive. What we'd want is an opaque subtype of number, i.e. `export opaque type Discount: number = number`